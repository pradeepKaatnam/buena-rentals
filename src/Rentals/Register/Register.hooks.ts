import { useEffect, useReducer } from 'react';
import { IRegisterState } from './Register.models';
import { IRegisterActions, RegisterActions } from './Register.actions';
import { registerReducer } from './Register.reducer';
import { IRegisterService, RegisterService } from './Register.service';

export const useInit = (
): {
  state: IRegisterState;
  actions: IRegisterActions;
} => {
  const [state, dispatch] = useReducer(registerReducer, {
    financialInformation: {},
    personalInformation: {},
  } as IRegisterState);
  
  const service: IRegisterService = new RegisterService();
  const actions: IRegisterActions = new RegisterActions(dispatch, service);
  useEffect(() => {}, []);
  return { state, actions };
};
