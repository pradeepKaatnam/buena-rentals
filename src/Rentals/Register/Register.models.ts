export interface IRegisterProps {

}

export interface IPersonalInformationProps {
    formData: IPersonalInformation;
    onFormChange: (formType: string, key: string, value: string) => void;
}

export interface IFinancialInformationProps {
    formData: IFinancialInformation;
    onFormChange: (formType: string, key: string, value: string) => void;
}   

export interface IReviewProps {
    personalInformation: IPersonalInformation;
    financialInformation: IFinancialInformation;
}

export interface IPersonalInformation {
    firstName: string;
    lastName: string;
    addressLine1: string;
    addressLine2?: string;
    email: string;
    nationality: string;
}

export interface IFinancialInformation {
    employmentStatus: string;
    employer: string;
    occupation: string;
    workDuration: string;
    salaryIndication: number;
}

export interface IRegisterState {
    personalInformation: IPersonalInformation;
    financialInformation: IFinancialInformation;
    isSubmitting: boolean;
    isSubmitted: boolean;
    errorMessage: string;
}
