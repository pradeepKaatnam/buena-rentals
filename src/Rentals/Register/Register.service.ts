export interface IRegisterService {
}
  
export class RegisterService implements IRegisterService {
}