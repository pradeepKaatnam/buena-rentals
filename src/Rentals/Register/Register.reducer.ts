import { IRegisterDispatchActions, REGISTER_ACTIONS } from "./Register.actions";
import { IRegisterState } from "./Register.models";

export const registerReducer = (state: IRegisterState, action: IRegisterDispatchActions): IRegisterState => {
    switch (action.type) {
        case REGISTER_ACTIONS.UPDATE_PERSONAL_INFO_ENTRY: {
          return {
            ...state,
            personalInformation: {
              ...state.personalInformation,
              [action.data.key]: action.data.value
            },
          };
        }
        case REGISTER_ACTIONS.UPDATE_FINANCIAL_INFO_ENTRY: {
          return {
            ...state,
            financialInformation: {
              ...state.financialInformation,
              [action.data.key]: action.data.value,
            },
          };
        }
        case REGISTER_ACTIONS.UPDATE_FINANCIAL_INFO: {
            return { ...state,  financialInformation: action.data };
        }
        case REGISTER_ACTIONS.UPDATE_PERSONAL_INFO: {
            return { ...state,  personalInformation: action.data };
        }
        case REGISTER_ACTIONS.UPDATE_API_FLOW: {
            return { ...state, ...action.data };
        }
        default: {
          throw new Error(`Unhandled action type`);
        }
      }
};
  