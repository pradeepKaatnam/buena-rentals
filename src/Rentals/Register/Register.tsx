import React from 'react';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Grid from '@mui/material/Grid2';
import Stack from '@mui/material/Stack';
import Step from '@mui/material/Step';
import StepLabel from '@mui/material/StepLabel';
import Stepper from '@mui/material/Stepper';
import Typography from '@mui/material/Typography';
import RentalsInfo from './RentalsInfo';
import PersonalInformation from './Forms/PersonalInformation';
import FinancialInformation from './Forms/FinancialInformation';
import Review from './Review';
import useStyles, { RegisterStyles } from './Register.styles';
import ChevronRightRoundedIcon from '@mui/icons-material/ChevronRightRounded';
import ChevronLeftRoundedIcon from '@mui/icons-material/ChevronLeftRounded';
import { useInit } from './Register.hooks';
import { validateFinancialInformation, validatePersonalInformation } from '../../common/utilities/form-utilities';
import { CircularProgress } from '@mui/material';

const Register: React.FC = () => {
    const steps = ['Personal Info', 'Financial Info', 'Review'];
    const [activeStep, setActiveStep] = React.useState(0);
    const { state, actions } = useInit();
    const classes = useStyles();

    const getStepContent = (step: number) => {
        switch (step) {
            case 0:
                return <PersonalInformation formData={state.personalInformation} onFormChange={onFormChange} />;
            case 1:
                return <FinancialInformation formData={state.financialInformation} onFormChange={onFormChange} />;
            case 2:
                return <Review personalInformation={state.personalInformation} financialInformation={state.financialInformation} />;
            default:
                return 'Unknown step';
        }
    }

    const handleBack = () => {
        setActiveStep((prevActiveStep) => prevActiveStep - 1);
    }

    const handleNext = async () => {
        if(activeStep === 0) {
            const isValidForm = validatePersonalInformation(state.personalInformation);
            if(!isValidForm) {
                actions.updatePersonalInformation(state.personalInformation);
                return;
            }
        } else if (activeStep === 1) {
            const isValidForm = validateFinancialInformation(state.financialInformation);
            if(!isValidForm) {
                actions.updateFinancialInformation(state.financialInformation);
                return;
            }
        }

        if(activeStep === 2) {
            await actions.register(state.personalInformation, state.financialInformation);
        }

        setActiveStep((prevActiveStep) => prevActiveStep + 1);
    }

    const onFormChange = (formType: string, key: string, value: string) => {    
        var formField = {
            value: value,
            error: value === '' ? 'This field is required' : ''
        };
        if(formType === 'personalInformation') {
            actions.updatePersonalInformationEntry(key, formField);
        } else {
            actions.updateFinancialInformationEntry(key, formField);
        }
    }

    return <>
        {state.loading && (
            <Box sx={RegisterStyles.loader}>
                <CircularProgress />
            </Box>
        )}
        <Grid container sx={{ height: { xs: '100%', sm: '100dvh' } }}>
          <Grid
            size={{ xs: 12, sm: 5, lg: 4 }}
            sx={RegisterStyles.gridLeftContainer}
          >
            <RentalsInfo/>
          </Grid>
          <Grid
            size={{ sm: 12, md: 7, lg: 8 }}
            sx={RegisterStyles.rightGridContainer}
          >
            <Box
              sx={RegisterStyles.rightGridBox}
            >
              <Box
                sx={RegisterStyles.stepperBox}
              >
                <Stepper
                  id="desktop-stepper"
                  activeStep={activeStep}
                  sx={{ width: '100%', height: 40 }}
                >
                  {steps.map((label) => (
                    <Step
                      sx={RegisterStyles.steps}
                      key={label}
                    >
                      <StepLabel>{label}</StepLabel>
                    </Step>
                  ))}
                </Stepper>
              </Box>
            </Box>
            <Card sx={{ display: { xs: 'flex', md: 'none' }, width: '100%' }}>
              <CardContent
                sx={{
                  backgroundColor: 'black',
                  color: 'white',
                }}
              >
                <RentalsInfo/>
              </CardContent>
            </Card>
            <Box
              sx={RegisterStyles.stepperBoxMobile}
            >
              <Stepper
                id="mobile-stepper"
                activeStep={activeStep}
                alternativeLabel
                sx={{ display: { sm: 'flex', md: 'none' } }}
              >
                {steps.map((label) => (
                  <Step
                    sx={RegisterStyles.steps}
                    key={label}
                  >
                    <StepLabel
                      sx={{ '.MuiStepLabel-labelContainer': { maxWidth: '70px' } }}
                    >
                      {label}
                    </StepLabel>
                  </Step>
                ))}
              </Stepper>
              {activeStep === steps.length ? (
                <Stack spacing={2} useFlexGap>
                  <Typography variant="h1">ðŸŽ‰</Typography>
                  <Typography variant="h5">Thank you for registering!</Typography>
                  <br/>
                  <Button
                    variant="contained"
                    sx={{ alignSelf: 'start', width: { xs: '100%', sm: 'auto' }, backgroundColor: 'black',}}
                  >
                    Go to rentals
                  </Button>
                </Stack>
              ) : (
                <React.Fragment>
                  {getStepContent(activeStep)}
                  <Box
                    sx={[
                      RegisterStyles.buttonsBox,
                      activeStep !== 0
                        ? { justifyContent: 'space-between' }
                        : { justifyContent: 'flex-end' },
                    ]}
                  >
                    {activeStep !== 0 && (
                      <Button
                        startIcon={<ChevronLeftRoundedIcon />}
                        onClick={handleBack}
                        variant="text"
                        sx={{ display: { xs: 'none', sm: 'flex' } }}
                        style={{ color: 'black' }}
                      >
                        Previous
                      </Button>
                    )}
                    <Button
                      variant="contained"
                      endIcon={activeStep < steps.length - 1 && <ChevronRightRoundedIcon /> }
                      onClick={handleNext}
                      sx={{ width: { xs: '100%', sm: 'fit-content', backgroundColor: 'black',
                        color: 'white' } }}
                    >
                      {activeStep === steps.length - 1 ? 'Submit' : 'Next'}
                    </Button>
                  </Box>
                </React.Fragment>
              )}
            </Box>
          </Grid>
        </Grid>
    </>;
};

export default Register;