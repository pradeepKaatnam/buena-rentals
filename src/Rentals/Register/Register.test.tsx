import React from 'react';
import { render } from '@testing-library/react';
import Register from './Register';
import { ThemeProvider, createTheme } from '@mui/material/styles';
import { IFinancialInformation, IPersonalInformation } from './Register.models';

// Mock the actions module
jest.mock('./Register.hooks', () => ({
    useInit: () => ({
        state: {
            financialInformation: {
                employer: { value: '', error: '' },
                employmentStatus: { value: '', error: '' },
                occupation: { value: '', error: '' },
                salaryIndication: { value: '', error: '' },
                workDuration: { value: '', error: '' },
              } as IFinancialInformation,
              personalInformation: {
                addressLine1: { value: '', error: '' },
                addressLine2: { value: '', error: '' },
                email: { value: '', error: '' },
                firstName: { value: '', error: '' },
                lastName: { value: '', error: '' },
                nationality: { value: '', error: '' },
              } as IPersonalInformation,
            loading: false,
        },
        actions: {
            updatePersonalInformation: jest.fn(),
            updateFinancialInformation: jest.fn(),
            updatePersonalInformationEntry: jest.fn(),
            updateFinancialInformationEntry: jest.fn(),
            register: jest.fn(),
        },
    }),
}));

// Mock the RegisterStyles module
jest.mock('./Register.styles', () => ({
    __esModule: true,
    default: () => ({
      root: 'root',
      // other mocked class names
    }),
    RegisterStyles: {
        loader: {},
        gridLeftContainer: {},
        rightGridContainer: {},
        rightGridBox: {},
        stepperBox: {},
        stepperBoxMobile: {},
        steps: {},
        buttonsBox: {},
    },
}));

describe('Register Component', () => {
    it('matches the snapshot', () => {
        const { asFragment } = render(
            <Register />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});