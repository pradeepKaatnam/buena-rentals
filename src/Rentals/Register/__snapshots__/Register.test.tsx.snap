// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-158jqj9-MuiGrid2-root"
  >
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-sm-5 MuiGrid2-grid-lg-4 css-f3i8j2-MuiGrid2-root"
    >
      <div
        class="bg-white text-black p-4 MuiBox-root css-ph8oxj"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-8s1fgi-MuiTypography-root"
        >
          Buena
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-alignJustify css-d4fp74-MuiTypography-root"
          style="--Typography-textAlign: justify;"
        >
          Our program, Buena Rentals, is dedicated to providing the best rental services in the industry. We offer a wide range of rental properties to suit all needs and budgets. Our team is committed to ensuring that our clients have a seamless and enjoyable rental experience.
        </p>
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-sm-12 MuiGrid2-grid-md-7 MuiGrid2-grid-lg-8 css-y7cztp-MuiGrid2-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiStepper-root MuiStepper-horizontal css-rkn5dl-MuiStepper-root"
            id="desktop-stepper"
          >
            <div
              class="MuiStep-root MuiStep-horizontal css-nd0xiu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal css-ascpo7-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer Mui-active css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-15rfbsb-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      dominant-baseline="central"
                      text-anchor="middle"
                      x="12"
                      y="12"
                    >
                      1
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-active css-1hv8oq8-MuiStepLabel-label"
                  >
                    Personal Info
                  </span>
                </span>
              </span>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-1qv8q39-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-horizontal css-nd0xiu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer Mui-disabled css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-15rfbsb-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      dominant-baseline="central"
                      text-anchor="middle"
                      x="12"
                      y="12"
                    >
                      2
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-disabled css-1hv8oq8-MuiStepLabel-label"
                  >
                    Financial Info
                  </span>
                </span>
              </span>
            </div>
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-1qv8q39-MuiStepConnector-line"
              />
            </div>
            <div
              class="MuiStep-root MuiStep-horizontal css-nd0xiu-MuiStep-root"
            >
              <span
                class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
              >
                <span
                  class="MuiStepLabel-iconContainer Mui-disabled css-vnkopk-MuiStepLabel-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-15rfbsb-MuiSvgIcon-root-MuiStepIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="12"
                    />
                    <text
                      class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                      dominant-baseline="central"
                      text-anchor="middle"
                      x="12"
                      y="12"
                    >
                      3
                    </text>
                  </svg>
                </span>
                <span
                  class="MuiStepLabel-labelContainer css-1vyamtt-MuiStepLabel-labelContainer"
                >
                  <span
                    class="MuiStepLabel-label Mui-disabled css-1hv8oq8-MuiStepLabel-label"
                  >
                    Review
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-qfnmc0-MuiPaper-root-MuiCard-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiCardContent-root css-ynms3u-MuiCardContent-root"
        >
          <div
            class="bg-white text-black p-4 MuiBox-root css-ph8oxj"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-8s1fgi-MuiTypography-root"
            >
              Buena
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignJustify css-d4fp74-MuiTypography-root"
              style="--Typography-textAlign: justify;"
            >
              Our program, Buena Rentals, is dedicated to providing the best rental services in the industry. We offer a wide range of rental properties to suit all needs and budgets. Our team is committed to ensuring that our clients have a seamless and enjoyable rental experience.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel css-qx00f7-MuiStepper-root"
          id="mobile-stepper"
        >
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-190y1a4-MuiStep-root"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-qasq96-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-active MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-15rfbsb-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    1
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-active MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label"
                >
                  Personal Info
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-190y1a4-MuiStep-root"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-1gg6q23-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-1qv8q39-MuiStepConnector-line"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-qasq96-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-disabled MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-15rfbsb-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    2
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-disabled MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label"
                >
                  Financial Info
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-190y1a4-MuiStep-root"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-1gg6q23-MuiStepConnector-root"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-1qv8q39-MuiStepConnector-line"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-qasq96-MuiStepLabel-root"
            >
              <span
                class="MuiStepLabel-iconContainer Mui-disabled MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-15rfbsb-MuiSvgIcon-root-MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                    dominant-baseline="central"
                    text-anchor="middle"
                    x="12"
                    y="12"
                  >
                    3
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer MuiStepLabel-alternativeLabel css-1vyamtt-MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiStepLabel-label Mui-disabled MuiStepLabel-alternativeLabel css-1hv8oq8-MuiStepLabel-label"
                >
                  Review
                </span>
              </span>
            </span>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-3 css-2vr48m-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 css-cl2elj-MuiGrid2-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-s6km2v-MuiFormLabel-root"
              for="first-name"
            >
              First name
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
              required=""
            >
              <input
                autocomplete="first name"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1sh6wjw-MuiInputBase-input-MuiOutlinedInput-input"
                id="first-name"
                name="firstName"
                placeholder="John"
                required=""
                type="name"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-gpjuor"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-6 css-cl2elj-MuiGrid2-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-s6km2v-MuiFormLabel-root"
              for="last-name"
            >
              Last name
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
              required=""
            >
              <input
                autocomplete="last name"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1sh6wjw-MuiInputBase-input-MuiOutlinedInput-input"
                id="last-name"
                name="lastName"
                placeholder="Snow"
                required=""
                type="last-name"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-gpjuor"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-63msfb-MuiGrid2-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-s6km2v-MuiFormLabel-root"
              for="address1"
            >
              Address line 1
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
              required=""
            >
              <input
                autocomplete="shipping address-line1"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1sh6wjw-MuiInputBase-input-MuiOutlinedInput-input"
                id="address1"
                name="addressLine1"
                placeholder="Street name and number"
                required=""
                type="address1"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-gpjuor"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-63msfb-MuiGrid2-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-s6km2v-MuiFormLabel-root"
              for="address2"
            >
              Address line 2
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
              required=""
            >
              <input
                autocomplete="shipping address-line2"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1sh6wjw-MuiInputBase-input-MuiOutlinedInput-input"
                id="address2"
                name="addressLine2"
                placeholder="Apartment, suite, unit, etc. (optional)"
                required=""
                type="address2"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-gpjuor"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 css-adu8q4-MuiGrid2-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-s6km2v-MuiFormLabel-root"
              for="city"
            >
              Email
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
              required=""
            >
              <input
                autocomplete="City"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1sh6wjw-MuiInputBase-input-MuiOutlinedInput-input"
                id="city"
                name="email"
                placeholder="New York"
                required=""
                type="city"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-gpjuor"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 css-adu8q4-MuiGrid2-root"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-s6km2v-MuiFormLabel-root"
              for="state"
            >
              Nationality
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-sizeSmall css-j882ge-MuiInputBase-root-MuiOutlinedInput-root"
              required=""
            >
              <input
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1sh6wjw-MuiInputBase-input-MuiOutlinedInput-input"
                id="nationality"
                name="nationality"
                required=""
                type="nationality"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-gpjuor"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1bodhlx"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-kb5ssh-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Next
            <span
              class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-1gulhci-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-20bmp1-MuiSvgIcon-root"
                data-testid="ChevronRightRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.29 6.71c-.39.39-.39 1.02 0 1.41L13.17 12l-3.88 3.88c-.39.39-.39 1.02 0 1.41s1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41L10.7 6.7c-.38-.38-1.02-.38-1.41.01"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
